storage/disk/disk.go

```diff
@@ -0,0 +1,420 @@
+// Copyright 2021 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+// Package disk provides disk-based implementation of the storage.Store
+// interface.
+//
+// The disk.Store implementation uses an embedded key-value store to persist
+// policies and data. Policy modules are stored as raw byte strings with one
+// module per key. Data is mapped to the underlying key-value store with the
+// assistance of caller-supplied "partitions". Partitions allow the caller to
+// control the portions of the /data namespace are mapped to individual keys.
+// Operations that span multiple keys (e.g,. a read against the entirety of
+// /data) are more expensive than reads that target a specific key because the
+// storage layer has reconstruct the object from individual key-value pairs and
+// page all of the data into memory. By supplying partitions that align with
+// lookups in the policies, callers can optimize policy evaluation.
```

Comment:
Agreed this could be come a common need. I could imagine tooling to recommend partitions based on a policy and/or tooling to lint policies and identify inefficient statements. Both could be valuable. I'll start a list of improvements that we can address in follow up PRs.